### ISSUE TITLE
Content of wrong feed is shown

### ISSUE THREAD

**Issue by Nudin** [lwindolf/liferea#884]
From time to time the content of a news entry from an entirely different is shown instead of the correct one. See below for an example: The title and author if from a feed of the commits of a git repo, while the content is from a completely different feed (it's in german)…

![Bildschirmfoto](https://user-images.githubusercontent.com/509205/93937644-68621900-fd28-11ea-9030-dfb7b2aecd83.png)

**Comment by Nudin** [comment]
Also: sometimes the audio-recording attached to one news item is shown on anther one (again different feed).

**Comment by rich-coe** [comment]
I not sure about the feed, but on git, it will sometimes show ''wrong'' information, like the author/comment of the last commit on a branch when a new branch is created.

**Comment by Nudin** [comment]
> I not sure about the feed, but on git, it will sometimes show ''wrong'' information, like the author/comment of the last commit on a branch when a new branch is created.

It's not a bug in the feed. The content shown isn't from github and it occurs with all kinds of feeds from blogs, newpages, etc.

**Comment by rich-coe** [comment]
Your going to have to point to a public feed that exhibits the problem, that is reproducible. 
What I'm trying to explain is that complaining about the content of feed coming from a git repo is not conclusive evidence of a problem.

**Comment by Leiaz** [comment]
@rich-coe That is an actual bug. It happened to me, but only once, on one item only. As I run master, plus sometimes other changes or temporary tests, and I couldn't reproduce it, I didn't search further.

@Nudin Can you reproduce it reliably ? Which version of Liferea are you using ?

**Comment by Nudin** [comment]
Yes it is reproducable in the sense that the item stays 'brocken' also after restating liferea. I'm using version 1.13.2. At least the variant of the bug with the audio file already happened several months ago.

**Comment by Nudin** [comment]
I looked into the liferea.db:
- In the 'item' table, I could find the item by it's title and it's description-field contains the correct content.
- I could however not find where the (wrong) content is stored.

**Comment by Leiaz** [comment]
You are using "extract full content" on the misplace article content too ?

I think I know the issue : the metadata table still contains the metadata of deleted items. At least that what seams to happen.
```
sqlite> select count(distinct metadata.item_id) from metadata left join items on metadata.item_id = items.item_id where items.item_id is null;
2458
```
Full content is in the metadata table.

**Comment by Nudin** [comment]
> You are using "extract full content" on the misplace article content too ?

I'm using it on the feed that the content is from – but not on the feed the header is from / where the content is shown.

> I think I know the issue : the metadata table still contains the metadata of deleted items. At least that what seems to happen.
> 
> Full content is in the metadata table.

Yes, can confirm that. The missing deletion combined with the reuse of ids cause the issue.

**Comment by Leiaz** [comment]
The `item_removal` trigger seems to be working, so I don't know how those metadata rows escaped deletion. Maybe a crash during update ?

**Comment by lwindolf** [comment]
Should be fixed with #898

**Comment by lwindolf** [comment]
@Nudin Can you please retest with master?

**Comment by Leiaz** [comment]
@lwindolf This is tricky to test because for the affected articles, it is too late. They won't be fixed by the patch. The remaining metadata item now correspond to an actual item, it won't be cleaned.

The change to item_id generation will help by not reusing ids too fast, so in case there are leftover "orphan" metadata items they will have more time to be removed from the database.
I'm not sure this will be fixed by #887 because I don't know how those metadata items escaped deletion by the trigger in the first place.

**Comment by lwindolf** [comment]
@Nudin @Leiaz Right now I'm just hoping it will work ;-)

**Comment by Nudin** [comment]
This bug was fixed for 2½ years – but with 1.14.x it seems to have returned. Should I create a new issue or should we reopen this.
